lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -194,16 +233,18 @@ def test_normal_created():
                 tarball,
             ],
         )
-        assert result.exit_code == 0, result.stderr
+        assert result.exit_code == exit_code, result.stderr
         assert result.stdout == ""
+        assert result.stderr.strip() == "" if not message else message["message"]
```

Comment:
I don't think the assertion at line 238 is doing what you want.  The Python [operator precedence rules](https://docs.python.org/3/reference/expressions.html#operator-precedence) put the ternary operation way down the list, so if `message` is false-y, then the assertion will evaluate `result.stderr.strip() == ""` and otherwise it will evaluate `bool(message["message"])` (which is probably always truish).  So, you need to add some parentheses:
```python
        assert result.stderr.strip() == ("" if not message else message["message"])
```

Reply:
With the assertion fixed, there are two failing tests: in fact, I wondered in passing :-) why the tests were passing and then promptly forgot because of other failures - and didn't remember to go back and check again. Sigh...